recipes/libsecret/all/conanfile.py

```diff
@@ -0,0 +1,86 @@
+from conans import ConanFile, Meson, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class LibsecretConan(ConanFile):
+    name = "libsecret"
+    description = "A library for storing and retrieving passwords and other secrets"
+    topics = ("conan", "libsecret", "gobject", "password", "secret")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://wiki.gnome.org/Projects/Libsecret"
+    license = "LGPL-2.1"
+    generators = "pkg_config"
+
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def validate(self):
+        if self.settings.os == "Windows":
+            raise ConanInvalidConfiguration(
+                "libsecret recipe is not yet compatible with Windows."
+            )
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def requirements(self):
+        self.requires("glib/2.67.2")
+
+    def build_requirements(self):
+        self.build_requires("meson/0.56.2")
+        self.build_requires("pkgconf/1.7.3")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename(self.name + "-" + self.version, self._source_subfolder)
+
+    def _configure_meson(self):
+        meson = Meson(self)
+        defs = {}
+        defs["introspection"] = "false"
+        defs["manpage"] = "false"
+        defs["gtk_doc"] = "false"
+        defs["gcrypt"] = "false"
+        meson.configure(
+            defs=defs,
+            build_folder=self._build_subfolder,
+            source_folder=self._source_subfolder,
+            pkg_config_paths=[self.install_folder],
+        )
+        return meson
+
+    def build(self):
+        with tools.environment_append(tools.RunEnvironment(self).vars):
+            meson = self._configure_meson()
+            meson.build()
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        with tools.environment_append({"PKG_CONFIG_PATH": self.install_folder}):
```

Comment:
Is this environment needed when the `Meson` ctor already got the `pkg_config_paths=[self.install_folder]` argument? If so, seems that it is a bug/inconsitency in the `Meson` helper.

I see this pattern in gtk, graphene and other recipes using pkgconfig.

Reply:
IIRC, it is necessary because conan does not forward the `pkg_config_paths` to `meson.install()`

Reply:
That's right, I see, it is used only in `configure` (which I meant by ctor).

How do you think, should we add this to `build` and `package`? It will remove a lot of boilerplate in recipes.